### ISSUE TITLE
[HELP] RGPLua Crashes Finale

### ISSUE THREAD

**Issue by SD308** [finale-lua/lua-scripts#542]
**What do you need help with?**

When I tried to install it, it was fine, until it asked me which folder(s) for it to look through for plug-ins.  It accepted my request (the entire plug-ins folder), then told me to close finale. I did so, and upon reopening it, it immediately crashes. Finale runs when I do not have RGP in the plugins folder, but then I can't use my Lua scripts.

**What have you tried?**

I have tried downloading a fresh version of RGP and replacing it. I tried an uninstall and immediately reinstall it. I attempted to see if there was a way to "reset" RGP's search to something else.

A few details I should mention: I am running Windows 11 Pro (Build 22621), and have Finale 27.

**Comment by rpatters1** [comment]
I've been tracking what appears to be the same issue with another user. I traced the code down to a call to `CryptDecrypt` in wincrypt.h. It appears to be failing there, and I am getting no diagnostic info. The program simply aborts. @asherber do you have any ideas of how to proceed? (You know more about Win than I do.) Could some windows installs be missing a component? Is there a way to check for messages in the system logs?

**Comment by asherber** [comment]
Hmm. I would think that the crypto resources would be standard on all installs, since they're used for lots of things. @rpatters1 Do you have a sample app or sample code you could send with a minimal repro of the issue? Of course, I'm still on Windows 10, but at least maybe I can see what areas might be problematic.

Maybe also ask the mailing list if anyone is successfully using RGPLua latest on Windows 11.

**Comment by rpatters1** [comment]
I can't reproduce it at all on my machine. I had to use debug printfs to trace it on a user's machine. He say he's using Win10, so it may not be related to Win11.

That said, I found a notice that wincrypt apis are deprecated, which made me wonder if there is an optional installer somewhere.

**Comment by asherber** [comment]
Deprecated, but not yet removed. These are part of the basic Win32 api, so I would expect them to be present.

**Comment by rpatters1** [comment]
I think I've fixed it. Fairly or not, Imma blame this on ChatGPT. But my code was calling CryptDecrypt with a null output buffer (to get the length). But that ain't how it works, because of course it's a Microsoft API so it's not going to be parallel to CryptEncrypt. That would make too much sense.

No, CryptDecrypt apparently takes as given that the decrypted text will be no longer than than the encrypted text. To be fair to ChatGPT, it also pointed out this discrepancy in my code after I asked it about it point blank.

So I think I've found the bug. I'll think about the best way to release a fix.
